### PR TITLE
[Enhancement] Support wild cards in the file paths for external tables.

### PR DESCRIPTION
**Pull request by fenganli**
Why I'm doing:
Wild cards in the file paths are not supported for external file tables.

What I'm doing:
Add an option called `enable_wildcards` for external file tables. If the option is enabled, then wild cards are supported in the file paths for external file tables.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by Smith-Cruise** [comment]
@GavinMar plz take a review

**Comment by stephen-shelby** [comment]
Is your main usage scenario file table?

**Comment by fenganli** [comment]
> Is your main usage scenario file table?

Yes.

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java
Comment (Smith-Cruise) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java:112]: why when expandWildCards=true needs to setFullPath

**Review by Smith-Cruise — COMMENTED** [review]
Add UT to test wild cards.
You can refer to `HiveRemoteFileIOTest.java`

**Comment by fenganli** [comment]
> Add UT to test wild cards. You can refer to `HiveRemoteFileIOTest.java`

Done

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java
Comment (fenganli) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveRemoteFileIO.java:112]: See the change in `fe/fe-core/src/main/java/com/starrocks/planner/FileTableScanNode.java`

Basically when `wildcards` is in tableLocation, we cannot simply concatenate the `tableLocation` + `fileName` to get the FullPath as `tableLocation` might contain wildcards. Thus we just pass down the fullPaths directly.

**Review by Smith-Cruise — APPROVED** [review]

**Review by GavinMar — COMMENTED** [review]
LGTM

**Review by GavinMar — APPROVED** [review]

**Comment by fenganli** [comment]
Thanks guys for approving the PR. However, the PR seems to require a `maintainer` approval? Could you guys help me find a `maintainer` to approve it? Thanks!

**Review by Youngwb — DISMISSED** [review]

**Comment by fenganli** [comment]
Sorry, could you guys please help me run the workflows? @Youngwb @GavinMar 

> If you are comfortable with running workflows on the pull request branch, return to the  Conversation tab, and under "Workflow(s) awaiting approval", click Approve and run.

**Comment by stephen-shelby** [comment]
> Sorry, could you guys please help me run the workflows? @Youngwb @GavinMar
> 
> > If you are comfortable with running workflows on the pull request branch, return to the  Conversation tab, and under "Workflow(s) awaiting approval", click Approve and run.

done

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by GavinMar — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/RemoteFileDesc.java
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/RemoteFileDesc.java:148]: check if it's null then print

File: fe/fe-core/src/main/java/com/starrocks/catalog/FileTable.java
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/FileTable.java:136]: what's your current blocking point if we don't support wild cards? if you have a partitioned table, you could set path to table location and enable the `enable_recursive_listing`.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FileTable.java:136]: oh, could you want to query multiple tables or partitions in a fileTable?

**Review by stephen-shelby — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by chaoyli** [comment]
@backport branch-3.1-cs

**Pull request merged**
